### PR TITLE
Enable selection between compass sensors on different buses

### LINKED ISSUE #2494: Mag HMC5883 on Revo not detected
### Board and Version
_Board name goes here. If you have a link where you've purchased the board - please include it_
CC3D Revo F4 : [https://librepilot.atlassian.net/wiki/spaces/LPDOC/pages/26968084/OpenPilot+Revolution](url)

_Version of INAV used goes here_
configurator 1.8.1 and firmware 1.8

_Use CLI command `version` and put its output here_

_If it's a custom-compiled firmware please mention this_

### Behavior
* _Describe the problem_

Since upgrade from configurator 1.7.3 and firmware 1.7.3 to configurator 1.8 and firmware 1.8, the magnetometer HMC5883 on the CC3D REVO is not detected.
Same Behavior with configurator 1.8 and firmware 1.7.3

* _Steps needed to reproduce the problem_

Upgrade the configurator to 1.8.1, goes to configuration tab and the field for the mag is set to NONE.
log: 
2017-11-22 @ 08:21:12 -- Running - OS: Windows, Chrome: 62.0.3202.94, Configurator: 1.8.1
2017-11-22 @ 08:22:33 -- Serial port successfully opened with ID: 1
2017-11-22 @ 08:22:33 -- MultiWii API version received - 2.1.0
2017-11-22 @ 08:22:33 -- Flight controller info, identifier: INAV, version: 1.8.0
2017-11-22 @ 08:22:33 -- Running firmware released on: Nov 1 2017 06:39:22
2017-11-22 @ 08:22:33 -- Board: REVO, version: 0
2017-11-22 @ 08:22:33 -- Unique device ID received - 0x3d00563535511031313931

If I select HMC5883, save and reboot, the configurator close the serial port after I want to connect it:

2017-11-22 @ 08:23:26 -- EEPROM saved
2017-11-22 @ 08:23:26 -- Device - Rebooting
2017-11-22 @ 08:23:27 -- Serial port successfully closed

then I do a reconnection and then the field show always NONE, and the configurator want an upgrade of the firmware whereas it's already the last revision:

2017-11-22 @ 08:23:30 -- Serial port successfully opened with ID: 2
2017-11-22 @ 08:23:31 -- MultiWii API version received - 2.1.0
2017-11-22 @ 08:23:31 -- Flight controller info, identifier: INAV, version: 1.8.0
2017-11-22 @ 08:23:31 -- Running firmware released on: Nov 1 2017 06:39:22
2017-11-22 @ 08:23:31 -- Board: REVO, version: 0
2017-11-22 @ 08:23:31 -- Unique device ID received - 0x3d00563535511031313931
2017-11-22 @ 08:23:32 -- Serial port successfully closed
2017-11-22 @ 08:23:33 -- You need to upgrade your firmware before you can use the Configuration tab.

* _Expected Results_

Mag HMC5883 detected as before

* _Actual Results_

mag HMC5883 not detected

### PR DESCRIPTION
**Pull request by digitalentity**
New option `mag_to_use` will allow to select betwen compass sensor on internal/external I2C bus (on boards where built-in compass is on separate I2C bus - only REVO so far).
Fixes #2494

### CODE REVIEW

File: src/main/drivers/compass/compass_ak8963.c [hunk 1]
```diff
@@ -164,7 +164,7 @@ static bool deviceDetect(magDev_t * mag)
 
 bool ak8963Detect(magDev_t * mag)
 {
-    mag->busDev = busDeviceInit(BUSTYPE_ANY, DEVHW_AK8963, 0, OWNER_COMPASS);
+    mag->busDev = busDeviceInit(BUSTYPE_I2C, DEVHW_AK8963, mag->magSensorToUse, OWNER_COMPASS);
```
Comment (MJ666) [inline src/main/drivers/compass/compass_ak8963.c:167]: If "BUSTYPE_I2C" instead of "BUSTYPE_ANY" is set here the AK8963 SPI compass will not be detected?

Comment (digitalentity) [reply src/main/drivers/compass/compass_ak8963.c:167]: @MJ666 oops, sorry. Will fix ASAP

**Comment by thomaskoul** [comment]
hi i have revo and ihave external mag i doing callibrate and no poshold why 
and with inav 1.8 working great

**Comment by thomaskoul** [comment]
and install 1.8 and do not callibrate compass all 0

**Pull request merged**